synapse/appservice/__init__.py

```diff
@@ -124,29 +125,23 @@ def _check_namespaces(self, namespaces):
                     raise ValueError(
                         "Expected bool for 'exclusive' in ns '%s'" % ns
                     )
-                if not isinstance(regex_obj.get("regex"), basestring):
+                regex = regex_obj.get("regex")
+                if isinstance(regex, basestring):
+                    regex_obj["regex"] = re.compile(regex)
```

Comment:
> A comment which is literally describing what a code line is doing

It isn't. This is a comment which literally describes what a code line is doing:
```py
# replace regex string with regex object
regex_obj["regex"] = re.compile(regex)
```

What I want to know is **why you're doing this**. Why don't we keep them as strings and then do it on the fly? Because that's needlessly slow compared to pre-compiling the regex. I don't care about the fact you're clobbering, I care that you fail to describe your intent.